File: ohjelmointikielet/zoo-java/Zoo.java

package fi.lab.zoo;

import java.util.*;

/**
 * @author Anni Karja
 */
public class Zoo {
    private Map<String, Animal> animalsMap = new HashMap<>();
    
    public Animal add(Animal a) {
        Animal temp = null;
        String name = a.getName();
        if(animalsMap.containsKey(name)) temp = animalsMap.get(name);
        animalsMap.put(name, a);
        return temp; 
    }
    
   public void remove(String name) {
       animalsMap.remove(name);
   }
   
   public void print() {
       System.out.println("---- Animals in Container ----");
       animalsMap.forEach((k,v) -> System.out.println("KEY: " + k + "\t VALUE: "+ v));
   }
   
   public void printSortedByName(){
       List<String> helper = new ArrayList<>();
       animalsMap.forEach((k, v) -> helper.add(k));
       Collections.sort(helper);
       System.out.println("---- Animals Sorted by Name ----");
       helper.forEach((e) -> System.out.println("KEY: " + e + "\t VALUE: " + animalsMap.get(e)));
   }
   
   public void printSortedByAge(){
       List<Animal> helper = new ArrayList<>();
       animalsMap.forEach((k, v) -> helper.add(v));
       helper.sort((Animal a1, Animal a2) -> a1.getBirth().compareAges(a2.getBirth()));
       System.out.println("---- Animals Sorted by Age ----");
       helper.forEach((e) -> System.out.println(e.getName()));
       }   
   }

File: ohjelmointikielet/zoo-cs/Zoo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zoo
{
    class Zoo
    {
        private Dictionary<string, Animal> animalsDict = new();

        public Animal add(Animal a)
        {
            Animal temp = null;
            string name = a.Name;
            if (animalsDict.ContainsKey(name))
            {
                temp = animalsDict[name];
            }

            animalsDict[name] = a;
            return temp; 
        }

        public void remove(string name)
        {
            animalsDict.Remove(name);
        }


        public void print()
        {
            Console.WriteLine("---- Animals in Container ----");
            foreach (var item in animalsDict) Console.WriteLine($"KEY: {item.Key} \t VALUE: {item.Value}");
        }

        public void printSortedByName()
        {
            List<String> helper = new();
            foreach (var item in animalsDict) helper.Add(item.Key);
            helper.Sort();
            Console.WriteLine("---- Animals Sorted by Name");
            foreach (var item in helper)
            {
                Console.WriteLine($"KEY: {item} \t VALUE: {animalsDict[item]}");
            }
        }

        public void printSortedByAge()
        {
            List<Animal> helper = new();
            foreach (var item in animalsDict) helper.Add(item.Value);
            helper.Sort(delegate (Animal a1, Animal a2)
            {
                 return a1.Birth.compareAges(a2.Birth);
            });
            Console.WriteLine("---- Animals Sorted by Age ----");
            foreach (var item in helper)
            {
                Console.WriteLine($"{item.Name}");
            }
        }
    }
}
